static/css/base/common.less

```diff
@@ -116,3 +116,27 @@ div.head {
 a.nostyle {
   color: inherit;
 }
+
+.loading-gradient {
+  // stylelint-disable-next-line color-no-hex
+  background: linear-gradient(to right, #dddddd 8%, #eeeeee 18%, #dddddd 33%);
+  background-size: 1000px 100%; /* Make the gradient wider than the element */
+  animation-name: shimmer;
+  animation-duration: 5s;
+  animation-timing-function: linear;
+  animation-iteration-count: 10;
+  border-radius: 4px;
+}
+
+.opacity-0 {
```

Comment:
In order to make this work in a non-JS environment, I think we can remove this class and instead make it `.loading-gradient img` which has `opacity: 0`. Then I think the specificity might be high enough that we don't need to specify `!important`. That'll allow us to have a rule in `head.html`'s `<noscript>` block, which sets `opacity: 1 !important` to override this for non-JS clients.

```css
.loading-gradient img {
    opacity: 0;
}
```